base/db/R/utils_db.R

```diff
@@ -187,100 +228,96 @@ db.exists <- function(params, write=TRUE, table=NA) {
 
   #check table's privilege about read and write permission
   user.permission <<- tryCatch({
-    invisible(db.query(paste0("select privilege_type from information_schema.role_table_grants where grantee='",params$user,"' and table_catalog = '",params$dbname,"' and table_name='",table,"'"), con))
+    invisible(db.query(
+      paste0("SELECT privilege_type FROM information_schema.role_table_grants ",
+             "WHERE grantee='", params$user,
+             "' AND table_catalog = '", params$dbname,
+             "' AND table_name='", table, "'"),
+      con
+    ))
   }, error = function(e) {
     PEcAn.logger::logger.error("Could not query database.\n\t", e)
     db.close(con)
     invisible(NULL)
   })
 
-  if (!is.na(table)){
-    read.perm = FALSE
-    write.perm = FALSE
+  # If table is NA, this is just a generic check for database access,
+  # so we're done!
+  if (is.na(table)) return(invisible(TRUE))
 
-    # check read permission
-    if ('SELECT' %in% user.permission[['privilege_type']]) {
-      read.perm = TRUE
-    }
+  # We're enquiring about permissions related to a specific table, so
+  # need to do more here.
+  read.perm <- FALSE
+  write.perm <- FALSE
 
-    #check write permission
-    if ('INSERT' %in% user.permission[['privilege_type']] &&'UPDATE' %in% user.permission[['privilege_type']] ) {
-      write.perm = TRUE
-    }
+  # check read permission
```

Comment:
The code here is the same, just done in a different order that, IMHO, makes more sense: First, do the complete check of read permissions (and return early if that fails). Then, check all the write permissions. (Before, they were interspersed).